src/lib/utils/socket/socket.cpp

```diff
@@ -118,7 +118,7 @@ class Asio_Socket final : public OS::Socket {
       void check_timeout() {
          if(m_tcp.is_open() && m_timer.expires_at() < std::chrono::system_clock::now()) {
             boost::system::error_code err;
-            m_tcp.close(err);
+            BOTAN_UNUSED(m_tcp.close(err));
```

Comment:
Isn't that a bit daring? If we ever changed `BOTAN_UNUSED` in a way that the enclosed expression isn't executed, this would become a bug. Does this fix it as well?:

```suggestion
            [[maybe_unused]] m_tcp.close(err);
```

Reply:
Ah better, yes